### PR TITLE
Support webhook urls with missing threadkey parameter

### PR DESCRIPTION
**Pull request by adamnfish**
NOTE: this includes the changes in the [new-inline-threads](https://github.com/guardian/actions-prnouncer/pull/20) PR

## What does this change?

Allows teams to provide a `{threadKey}` placeholder if they want to, but also supports cases where this is not included.

I am led to believe this sometimes casuses problems, when teams forget the parameter.

I'd suggest we officially deprecate the placeholder parameter from here, so that there is a single recommended approach. There's no reference to this placeholder in the README or elsewhere, so I've not made any documentation changes.

## How to test

I've added tests for this logic, but we'll also need to make sure that the existing behaviour still works. I do not have the secrets for testing at my fingertips, so if someone wants to pair with me on testing this locally I'd be grateful!

## How can we measure success?

Existing configurations should continue to work unchanged, and we might consider updating the webhook URL for one or more of the existing configurations, removing the deprecated threadKey placholder.

## Have we considered potential risks?

Testing this change locally would mitigate the risk of breaking existign configurations.

### CODE REVIEW

File: src/google.rs [hunk 1]
```diff
@@ -32,3 +54,44 @@ impl GoogleChatMessage {
...
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_build_threaded_webhook_with_placeholder() {
+        let result = GoogleChatMessage::build_webhook_url(&String::from("https://example.com/ABCDEF?threadKey={threadKey}"), &String::from("1234"));
+        match result {
+            Ok(url) => assert_eq!(url, String::from("https://example.com/ABCDEF?messageReplyOption=REPLY_MESSAGE_FALLBACK_TO_NEW_THREAD&threadKey=1234")),
+            Err(_) => assert!(false),
+        }
```
Comment (AshCorr) [inline src/google.rs:68]:
```suggestion
        assert_eq!(url, Ok(String::from("https://example.com/ABCDEF?messageReplyOption=REPLY_MESSAGE_FALLBACK_TO_NEW_THREAD&threadKey=1234")))
```

Is it possible to assert on a result perhaps?

**Review by AshCorr — APPROVED** [review]
Awesome :)

File: src/google.rs (hunk 1 above)
Comment (adamnfish) [reply src/google.rs:68]: That isn't possible (Result doesn't support equality checks), but `Result::unwrap` will work well in tests.

**Comment by adamnfish** [comment]
I've addressed both of these points, thank you!

**Comment by adamnfish** [comment]
We ran through testing this locally before merging 👍

**Pull request merged**